gtsfm/evaluation/compare_metrics.py

```diff
@@ -0,0 +1,81 @@
+"""Script to log metrics for comparison across SfM pipelines (GTSfM, COLMAP, etc.) as a JSON.
+
+Authors: Jon Womack
+"""
+import os
+import json
+from typing import Dict
+
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+
+
+def compare_metrics(txt_metric_paths: Dict[str, str], json_path: str) -> None:
+    """
+    Produces a json file containing data across SfM pipelines formatted as a GTSfMMetricsGroup.
+    Args:
+        txt_metric_paths: a list of paths to directories containing: cameras.txt, images.txt, and points3D.txt files
+    """
+    gtsfm_metrics = []
+    for pipeline_name in txt_metric_paths.keys():
+        # Add 3D point information
+        fpath = os.path.join(txt_metric_paths[pipeline_name], "points3D.txt")
+        with open(fpath, "r") as f:
+            data = f.readlines()
+            points = data[3:]
+            track_lengths = []
+            for point in points:
+                track_lengths.append(len(point.split()[8:]) / 2)
```

Comment:
its not clear to the reader how this is parsed, could you explain the format of the file being parsed in a comment?

Reply:
This reader only parses x, y, z, r, g, and b for each point, not information about each point's corresponding track.
Similarly, the reader for images doesn't parse each images 2D point information i.e. POINTS2D[] as (X, Y, POINT3D_ID).

The script now uses the camera parser (it is compatible as is).

Should I modify the readers to parse this additional information and the functions currently using these readers to disregard the additional information?

Reply:
Hi @womackj1, let's go ahead and use the official COLMAP python loader then. @travisdriver already wrote the converter to GtsfmData here:
https://github.com/borglab/gtsfm/blob/master/gtsfm/loader/astronet_loader.py#L76

so I propose we move his `colmap2gtsfm` function to `utils/io.py` and read off the tracks out of the `GtsfmData` using the helper methods we already have